packages/services/src/kernel/default.ts

```diff
@@ -235,11 +239,9 @@ export class KernelConnection implements Kernel.IKernelConnection {
     if (this._specPromise) {
       return this._specPromise;
     }
-    this._specPromise = KernelSpecAPI.getSpecs(this.serverSettings).then(
-      specs => {
-        return specs.kernelspecs[this._name];
-      }
-    );
+    this._specPromise = this._kernelSpecAPIClient.get().then(specs => {
+      return specs.kernelspecs[this._name];
+    });
```

Comment:
Should this one be replaced too?

https://github.com/jupyterlab/jupyterlab/blob/9749b07d54b4ac7622e617f1afdce770849c6f92/packages/services/src/kernelspec/manager.ts#L119-L120

Otherwise, we would could get discordant results, right?

Reply:
Should be fixed in https://github.com/jupyterlab/jupyterlab/pull/17395/commits/c34963e7d0b953801e523b371bb5a265bd1b5ee4